dotCMS/src/main/java/com/dotcms/visitor/filter/servlet/VisitorFilter.java

```diff
@@ -72,9 +72,17 @@ public void init(FilterConfig config) throws ServletException {
     public void doFilter(final ServletRequest req, final ServletResponse res, final FilterChain chain)
             throws IOException, ServletException {
 
+        // BYPASS HEALTH CHECK ENDPOINTS - prevent database access for K8s probes
+        final HttpServletRequest request = (HttpServletRequest) req;
+        final String servletPath = request.getServletPath();
+        if (HealthCheckEndpointUtil.isHealthCheckEndpoint(servletPath)) {
+            // Skip visitor filtering entirely for health endpoints - no database access
+            chain.doFilter(req, res);
+            return;
+        }
+
```

Comment:
Isn't there a more elegant way to bypass all these filters without dropping this conditional codeblock in all of them?
Maybe add it to its own filter and run it before these?  Not sure.

Reply:
Yes I agree and wanted to go in and fix this better,  but did not want to include other refactoring in this PR.  I intended on adding another issue and PR to follow up on this,   This change I felt gave the least impact to any existing functionality,  but I think there needs to be a little review on what exactly is being checked in the filters for every request and see if we can rely on caching before any attempt to connect.

Reply:
Adding its own filter at the start was one option to go at the start was one option I thought after this quick fix,  and I may be able to refactor pretty easily as it is cleaner.

Reply:
yeah, feels like it merits it. 👍

Reply:
I have had to rethink this solution as trying to change the filter chain used dynamically does not work properly.  There are 2 real solutions, 
1. create a request wrapper that identifies whether a filter should be excluded or not and change the Filters to respect this wrapper ( This has more impact on the other filters than the initial targeted change I made though )
2. I have to just order the Health filter in the web.xml after the "ESSENTIAL_FILTERS" but before the non essential ones and allow the filter to skip all remaining filters in the chain. 

For the time being I am going to go for 2 that will cover our immediate concerns of limited changes to the existing filter code, while ensuring we do not skip the known filters we know we should run.